terraform/modules/fargate_autoscaling/firewall.tf

```diff
@@ -1,81 +1,112 @@
+  security_group_id = aws_security_group.ecs_tasks.id
+  protocol          = "-1"
+  from_port         = 0
+  to_port           = 0
+  cidr_blocks       = ["0.0.0.0/0"]
+  type              = "egress"
+}
 
-# Open container port for egress and link with ECS Task security group
+# Egress rule for Load Balancer to ECS Tasks
 resource "aws_security_group_rule" "lb_task_egress" {
-  security_group_id        = length(aws_security_group.lb) == 1 ? aws_security_group.lb[0].id : var.load_balancer_security_group
+  security_group_id        = var.load_balancer_security_group == "" ? aws_security_group.lb[0].id : var.load_balancer_security_group
   from_port                = var.container_port
   to_port                  = var.container_port
   protocol                 = "tcp"
   type                     = "egress"
   source_security_group_id = aws_security_group.ecs_tasks.id
 }
 
-
-# Traffic to the ECS Cluster should only come from the ALB
-# Tasks should be able to communicate with any external resource
-resource "aws_security_group" "ecs_tasks" {
-  name        = substr("${var.project}-ecs-tasks${var.name_suffix}", 0, 64)
-  description = "Allow inbound access from the ALB only"
+# Security group for Batch instances
+resource "aws_security_group" "batch_instances" {
+  name        = substr("${var.project}-batch-instances${var.name_suffix}", 0, 64)
+  description = "Security group for AWS Batch jobs"
   vpc_id      = var.vpc_id
 
-  ingress {
-    protocol        = "tcp"
-    from_port       = var.container_port
-    to_port         = var.container_port
-    security_groups = length(aws_security_group.lb) > 0 ? [aws_security_group.lb[0].id] : [var.load_balancer_security_group]
-  }
-
-  egress {
-    protocol    = "-1"
-    from_port   = 0
-    to_port     = 0
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-
   tags = merge(
     {
-      Name = substr("${var.project}-ecs-tasks${var.name_suffix}", 0, 64)
+      Name = substr("${var.project}-batch-instances${var.name_suffix}", 0, 64)
     },
     var.tags
   )
 }
 
-resource "aws_security_group_rule" "lb" {
-  count             = var.load_balancer_security_group == "" ? 0 : 1
-  type              = "ingress"
-  from_port         = var.listener_port
-  to_port           = var.listener_port
-  protocol          = "tcp"
+# Egress rule for Batch instances
+resource "aws_security_group_rule" "batch_instances_egress" {
+  security_group_id = aws_security_group.batch_instances.id
+  protocol          = "-1"
+  from_port         = 0
+  to_port           = 0
   cidr_blocks       = ["0.0.0.0/0"]
-  security_group_id = var.load_balancer_security_group
+  type              = "egress"
 }
+
+# Allow Batch instances to communicate with ECS Tasks
+resource "aws_security_group_rule" "ecs_tasks_ingress_batch" {
+  security_group_id        = aws_security_group.ecs_tasks.id
+  protocol                 = "tcp"
+  from_port                = var.listener_port
+  to_port                  = var.listener_port
```

Comment:
Is `listener_port` being used for two meanings, what the load balancer should listen on and what the ECS tasks should listen on for batch jobs?  If so, can you update the comment on var.listener_port in variables.tf?  Or create a separate variable var.ecs_tasks_port with the same value 80?

Reply:
Ahh, I see. Yeah, the code and comments have diverged quite a bit. Will fix, thanks!